cacheops/query.py

```diff
@@ -248,6 +248,12 @@ def clone(self, klass=None, setup=False, **kwargs):
             clone._cloning = self._cloning - 1 if self._cloning else 0
             return clone
 
+    def cache_key_is_in_conj_sets(self, cache_key):
+        if not getattr(thread_local, 'fast_invalidation', False):
+            return True
+        dnfs_json = json.dumps(dnfs(self), default=str)
```

Comment:
If this call to `dnfs()` is a hot spot we can look at caching the DNFs for a query.